yaml/xyz/openbmc_project/Logging/AuditLog.interface.yaml

```diff
@@ -0,0 +1,43 @@
+description: >
+    This interface provides methods to extract the Linux audit logged events.
+methods:
+    - name: ParseAuditLog
+      description: >
+          Parses all previously recorded log events into JSON format. Returns a
+          read-only file descriptor to the parsed file. Entries are sorted
+          oldest to newest.
+      returns:
+          - name: logFd
+            type: unixfd
+            description: >
+                The read-only file descriptor of the JSON formatted log events.
+      errors:
+          - xyz.openbmc_project.Common.Error.InternalFailure
+          - xyz.openbmc_project.Common.File.Error.Write
+          - xyz.openbmc_project.Common.File.Error.Open
+
+    - name: GetAuditLog
```

Comment:
Does it simplify this by having 1 method and then having an optional parameter to defines the number of entries to return?

Reply:
I did think about defining just one method and using it for both operations. I decided not to because I thought documenting it and describing it would be overly complex due to differences in what is being returned:
-  Sort is opposite based on whether all entries are being returned or only a subset.
- Format of the JSON file is different between the methods. The full file download won't include the message registry. It is intended to be human-readable only.
- Long-term for upstream the full file download method seems less likely to get community acceptance. I thought it would be easier to keep these separated for trying to at least get some of this accepted upstream.